bitstore/controllers.py

```diff
@@ -160,3 +164,37 @@ def info(auth_token):
     except Exception as exception:
         logging.exception('Bad request (info)')
         return Response(status=400)
+
+
+def presign(auth_token, url, ownerid=None):
+    """Generates S3 presigned URLs if necessary
+    :param auth_token: authentication token from auth
+    :param ownerid: ownerid for dataset
+    :param url: url to check for sigend URL
+    """
+    s3 = get_s3_client()
+    try:
+        needs_signed_url = requests.get(url)
+        if needs_signed_url.status_code != 403:
+            return json.dumps({'url': url})
+        # Verify client, deny access if not verified
+        if ownerid is None:
+            return Response(status=400)
+        if not services.verify(auth_token, ownerid):
+            return Response(status=401)
+        parsed_url = requests.utils.urlparse(url)
```

Comment:
Better use the standard urllib.parse: https://docs.python.org/3/library/urllib.parse.html